feature/orchestrator/src/main/java/com/simprints/feature/orchestrator/ExecutionTracker.kt

```diff
@@ -3,36 +3,70 @@ package com.simprints.feature.orchestrator
 import androidx.lifecycle.Lifecycle
 import androidx.lifecycle.LifecycleEventObserver
 import androidx.lifecycle.LifecycleOwner
+import com.simprints.core.tools.time.TimeHelper
 import com.simprints.infra.logging.Simber
-import java.util.concurrent.atomic.AtomicBoolean
 import javax.inject.Inject
 import javax.inject.Singleton
 
 @Singleton
-internal class ExecutionTracker @Inject constructor() : LifecycleEventObserver {
+internal class ExecutionTracker @Inject constructor(
+    private val timeHelper: TimeHelper,
+    @ExecutorLockTimeoutSec private val executionTimeLimitSec: Int
```

Comment:
This seems like a bit of overengineering, it could have been a simple `const val` in this class, IMO.